sdk/python/kfp/v2/dsl/container_op_test.py

```diff
@@ -0,0 +1,57 @@
+"""Tests for kfp.v2.dsl.container_op"""
+import unittest
+
+from kfp.v2.dsl import container_op
+from kfp.v2.proto import pipeline_spec_pb2
+
+from google.protobuf import text_format
+from google.protobuf import json_format
+
+_EXPECTED_CONTAINER_WITH_RESOURCE = """
+resources {
+  cpu_limit: 1.0
+  memory_limit: 1.0
+  accelerator {
+    type: 'NVIDIA_TESLA_K80'
+    count: 1
+  }
+}
+"""
+
+# Shorthand for PipelineContainerSpec
+_PipelineContainerSpec = pipeline_spec_pb2.PipelineDeploymentConfig.PipelineContainerSpec
+
+
+class ContainerOpTest(unittest.TestCase):
+  def test_illegal_resource_setter_fail(self):
+    task = container_op.ContainerOp(name='test_task', image='python:3.7')
+    with self.assertRaisesRegex(TypeError, 'ContainerOp.container_spec '
+                                           'is expected to be'):
+      task.set_cpu_limit('1')
+
+  def test_illegal_label_fail(self):
+    task = container_op.ContainerOp(name='test_task', image='python:3.7')
+    task.container_spec = _PipelineContainerSpec()
+    with self.assertRaisesRegex(
+        ValueError, 'Currently add_node_selector_constraint only supports'):
+      task.add_node_selector_constraint('cloud.google.com/test-label',
+                                        'test-value')
+
+  def test_chained_call_resource_setter(self):
+    task = container_op.ContainerOp(name='test_task', image='python:3.7')
+    task.container_spec = _PipelineContainerSpec()
+    task.set_cpu_limit(
+        '1').set_memory_limit(
+        '1G').add_node_selector_constraint(
```

Comment:
AFAIK, putting parentheses around the method chain allows putting every method on its own line.

Reply:
Thanks! Done